nav2_recoveries/include/nav2_recoveries/recovery.hpp

```diff
@@ -173,12 +173,14 @@ class Recovery : public nav2_core::Recovery
       // TODO(orduno) #868 Enable preempting a Recovery on-the-fly without stopping
       if (action_server_->is_preempt_requested()) {
         RCLCPP_ERROR(
-          node_->get_logger(), "Received a preemption request for %s,"
-          " however feature is currently not implemented. Aborting and stopping.",
+          node_->get_logger(), "Received a preemption request for %s,",
           recovery_name_.c_str());
-        stopRobot();
-        action_server_->terminate_current();
-        return;
+        action_server_->accept_pending_goal();
```

Comment:
See comment above, I think for the is preemption or is cancelled, we just terminate the server running rather than sending a null goal.

Reply:
I think that makes sense.
Should I then revert the changes in recovery.hpp and just call halt on the tree node?
Or would you like to keep the code to accept a new goal if it's sent?

Reply:
Lets discuss in the comment thread above.